enterprise/server/backends/codesearch/codesearch.go

```diff
@@ -49,7 +54,61 @@ func (css *CodesearchService) Search(ctx context.Context, req *srpb.SearchReques
 	return css.client.Search(ctx, req)
 }
 
+func (css *CodesearchService) getGitRepoAccessToken(ctx context.Context, groupId, repoURL string) (string, error) {
+	ghas := css.env.GetGitHubAppService()
+	if ghas == nil {
+		return "", status.InternalError("GitHub App service is not configured")
+	}
+
+	gha, err := ghas.GetGitHubAppForOwner(ctx, repoURL)
+	if err != nil {
+		return "", status.UnavailableErrorf("could not get GitHub app for repo %q: %s", repoURL, err)
+	}
+
+	gitRepository := &tables.GitRepository{}
+	err = css.env.GetDBHandle().NewQuery(ctx, "hosted_runner_get_for_repo").Raw(`
+		SELECT *
+		FROM "GitRepositories"
+		WHERE group_id = ?
+		AND repo_url = ?
+	`, groupId, repoURL).Take(gitRepository)
+	if err != nil {
+		if db.IsRecordNotFound(err) {
+			return "", status.NotFoundErrorf("repo %q not found", repoURL)
+		}
+		return "", status.InternalErrorf("failed to look up repo %s: %s", repoURL, err)
+	}
+
+	token, err := gha.GetRepositoryInstallationToken(ctx, gitRepository)
+	if err != nil {
+		return "", status.UnavailableErrorf("could not get access token for repo %q: %s", repoURL, err)
+	}
+
+	return token, nil
+}
+
 func (css *CodesearchService) Index(ctx context.Context, req *inpb.IndexRequest) (*inpb.IndexResponse, error) {
```

Comment:
@vadimberezniker -- should this RPC be using your new capabilities check?

Reply:
This should be fine for now. I haven't merged those capabilities changes yet as I'm feeling more ambivalent about them now :sweat: